cpp/ql/test/library-tests/dataflow/fields/complex.cpp

```diff
@@ -33,8 +33,16 @@ void sink(int x)
 
 void bar(Bar &b)
 {
-  sink(b.f.a()); // flow (through `b1.f.setA` and `b3.f.setA`) [NOT DETECTED]
-  sink(b.f.b()); // flow (through `b2.f.setB` and `b3.f.setB`) [NOT DETECTED]
+  // The library correctly finds that the four `user_input` sources can make it
+  // to the `sink` calls, but it also finds some source/sink combinations that
+  // are impossible. Those false positives here are a consequence of how the
+  // shared data flow library overapproximates field flow. The library only
+  // tracks the head (`f`) and the length (2) of the field access path, and
+  // then it tracks that both `a_` and `b_` have followed `f` in _some_ access
+  // path somewhere in the search. That makes the library conclude that there
+  // could be flow to `b.f.a_` even when the flow was actually to `b.f.b_`.
+  sink(b.f.a()); // flow [FALSE POSITIVE through `b2.f.setB` and `b3.f.setB`]
+  sink(b.f.b()); // flow [FALSE POSITIVE through `b1.f.setA` and `b3.f.setA`]
```

Comment:
It would be nice to have a test case where the changes *do* get the correct result.

Reply:
When we discussed this, @jbj pointed out that there are correct results on these lines as well as false positive results (and he expects future changes to eliminate the FPs here).